libdnf/base/goal_elements.cpp

```diff
@@ -38,22 +38,24 @@ bool GoalJobSettings::resolve_strict(const libdnf::ConfigMain & cfg_main) {
             resolved = GoalUsedSetting::USED_FALSE;
             break;
     }
-    if (resolved == GoalUsedSetting::UNUSED) {
-        throw LogicError("Invalid 'strict' value in GoalJobSettings");
-    }
-    if (used_strict != GoalUsedSetting::UNUSED && resolved != used_strict) {
-        throw LogicError("GoalJobSettings::resolve_strict: 'strict' is already set to a different value");
-    }
+
+    // TODO this cannot happen, code above always sets USED_TRUE or USED_FALSE
+    libdnf_assert(resolved != GoalUsedSetting::UNUSED, "Invalid 'strict' value in GoalJobSettings");
+
+    libdnf_assert(used_strict == GoalUsedSetting::UNUSED || resolved == used_strict,
```

Comment:
I can be wrong, but the condition of assert was changed. It means a problem - it asserts in different condition.
I am thinking about the new error message. From my point of view we should provide as much information as we can, therefore the information about the place where the exception was raised  is nice to have.